netwayste/src/utils/ping.rs

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2020 The Conwayste Developers
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of  MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+use std::collections::VecDeque;
+use std::time::Instant;
+
+const PING_FILTER_DEPTH: usize = 12;
+
+pub struct PingFilter {
+    pub average_latency_ms: u128,
```

Comment:
Do we need `u128` here? I think this could be `u64`.
Also, it might be a good idea to wrap the number in an option, making the type be `Option<u64>`. That way, code that accesses this to get the ping time can easily tell the difference between it not being set yet and the ping time being `<1ms`.

Reply:
`u128` is used here because that's the return type for a `Duration`. I had thought about trimming it down to `u64` since it's a safe cast. Looks like you agree so I'll do so!

Also good idea about the `Option`.